src/System.Reflection/tests/Module/ModuleTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.CustomAttributesTests.Data;
+
+[module: Attr(77, name = "AttrSimple")]
+[module: Int32Attr(77, name = "Int32AttrSimple")]
+[module: Int64Attr((Int64)77, name = "Int64AttrSimple")]
+[module: StringAttr("hello", name = "StringAttrSimple")]
+[module: EnumAttr(MyColorEnum.GREEN, name = "EnumAttrSimple")]
+
+
+namespace System.Reflection.Tests
+{
+    public class ModuleTest
+    {
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(List<>))]
+        [InlineData(typeof(ModuleTest))]
+        public static void AssemblyTest(TypeInfo typeInfo)
+        {
+            Module module = typeInfo.Module;
+            Assert.Equal(typeInfo.Assembly, module.Assembly);
+        }
+
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(List<>))]
+        [InlineData(typeof(ModuleTest))]
+        public static void FullyQualifiedNameTest(TypeInfo typeInfo)
+        {
+            Module module = typeInfo.Module;
+            Assert.Contains(typeInfo.Assembly.GetName().Name, module.FullyQualifiedName, StringComparison.CurrentCultureIgnoreCase);
+            Assert.Contains(typeInfo.Assembly.GetName().Name, module.Name, StringComparison.CurrentCultureIgnoreCase);
+        }
+
+        [Theory]
+        [InlineData(typeof(Attr), 77, "AttrSimple")]
+        [InlineData(typeof(Int32Attr), 77, "Int32AttrSimple")]
+        [InlineData(typeof(Int64Attr), (Int64)77, "Int64AttrSimple")]
+        [InlineData(typeof(StringAttr), "hello", "StringAttrSimple")]
+        [InlineData(typeof(EnumAttr), MyColorEnum.GREEN, "EnumAttrSimple")]        
+        public static void CustomAttributesTest<CtorArg, NamedArg>(Type attrType, CtorArg expectedCtorValue, NamedArg expectedNamedValue)
+        {
+            Module module = typeof(ModuleTest).GetTypeInfo().Module;
+            var customAttrs = module.CustomAttributes.GetEnumerator();
+
+            CustomAttributeData attribute = null;
+            while (customAttrs.MoveNext())
+            {
+                if (customAttrs.Current.AttributeType.Equals(attrType))
+                {
+                    attribute = customAttrs.Current;
+                }
+            }
```

Comment:
Not a big deal, but any reason not to use foreach or LINQ for this kind of thing? e.g.
```C#
CustomAttributeData attribute = module.CustomAttributes.FirstOrDefault(a => a.AttributeType.Equals(attrType));
```
instead of:
```C#
var customAttrs = module.CustomAttributes.GetEnumerator();
CustomAttributeData attribute = null;
while (customAttrs.MoveNext())
{
    if (customAttrs.Current.AttributeType.Equals(attrType))
    {
        attribute = customAttrs.Current;
    }
}
```